nb2workflow/nbadapter.py

```diff
@@ -431,21 +431,32 @@ def execute(self, parameters, progress_bar = True, log_output = True, inplace=Fa
         return exceptions
 
     def _execute(self, parameters, progress_bar = True, log_output = True, inplace=False, callback_url=None, tmpdir_key=None):
-
         if not inplace :
             tmpdir = self.new_tmpdir(tmpdir_key)
             logger.info("new tmpdir: %s", tmpdir)
 
-            try:
-                output = subprocess.check_output(["git","clone", "--recurse-submodules", os.path.dirname(os.path.realpath(self.notebook_fn)), tmpdir])
-                # output = subprocess.check_output(["git","clone", "--depth", "1", "file://" + os.path.dirname(os.path.realpath(self.notebook_fn)), tmpdir])
-                logger.info("git clone output: %s", output)
-            except Exception as e:
-                logger.warning("git clone failed: %s, will attempt copytree", e)
-
-                os.rmdir(tmpdir)
-
-                shutil.copytree(os.path.dirname(os.path.realpath(self.notebook_fn)), tmpdir)
+            command = ["git","clone", "--recurse-submodules", os.path.dirname(os.path.realpath(self.notebook_fn)), tmpdir]
+            result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            git_output = result.stdout.decode('utf-8')
+            if result.returncode != 0:
+                logger.warning(f"git clone failed with code: {result.returncode}")
+                logger.warning("git clone output: %s", git_output)
+                try:
+                    os.rmdir(tmpdir)
+                except OSError: # directory is not empty
+                    # this error may occure if the repo was originally cloned by the different version of git utility
+                    # e.g. when repo is mounted with docker run -v
+                    if 'git-lfs' in git_output:
+                        # the known solution is just to initialize git-lfs in advance
+                        message = "git-lfs is not initialized"
+                    else:
+                        message = "unknown git clone error, see logs for details"
+                    raise Exception(message)
```

Comment:
Perhaps using the `GitPython` library you can better inspect the error returned from the clone command.

I used within the dispatcher, here a clone example:

https://github.com/oda-hub/dispatcher-app/blob/f8098edb7c19931fe41e4dc812814e27d54cfedd/cdci_data_analysis/analysis/renku_helper.py#L161C1-L175